File: Treinamento Calouros/2024/Turma 2/Contests/Simulado Primeira Fase OBI/poquemao.py
#https://neps.academy/br/exercise/8 - Po que mão

n = int(input())
x = int(input())
y = int(input())
z = int(input())

if x+y+z <= n:
    print(3)
else:
    if x+y<=n or x+z<=n or y+z<=n:
        print(2)
    else:
        if x<=n or y<=n or z<=n:
            print(1)
        else:
            print(0)

File: Treinamento Calouros/2024/Turma 2/Contests/Simulado Primeira Fase OBI/poquemao.cpp
//https://neps.academy/br/exercise/8 - Po que mão

#include <bits/stdc++.h>

using namespace std;

#define endl '\n'

int main(){
    ios_base::sync_with_stdio(false), cin.tie(nullptr);
    int n, v[3];
    cin >> n >> v[0] >> v[1] >> v[2];

    sort(v, v+3);

    int resp = 0;
    for(int i=0; i<3; i++){
        if(v[i]<=n){
            resp++;
            n-=v[i];
        }
    }
    cout << resp << endl;
    return 0;
}